src/App.js

```diff
@@ -1,7 +1,37 @@
 import "./App.css";
 import ProjectsPage from "./components/projects_page/projects_page";
-function App() {
-  return <ProjectsPage />;
-}
+import LoginPage from "./components/login_page/login_page";
+import { HashRouter, Route, Routes, Navigate } from "react-router-dom";
+import { useSelector } from "react-redux";
+import ProtectedRoutes from "./components/protected_routes";
+import { ROUTES } from "./routes";
+
+const App = () => {
+  const isAuthenticated = useSelector((state) => state.auth.isAuthenticated);
+  return (
+    <HashRouter>
+      <Routes>
+        <Route
+          path={ROUTES.LOGIN}
+          element={
+            !isAuthenticated ? <LoginPage /> : <Navigate to={ROUTES.HOME} />
+          }
+        />
+        <Route element={<ProtectedRoutes userAuth={isAuthenticated} />}>
+          <Route path={ROUTES.HOME} element={<ProjectsPage />} />
+        </Route>
+        <Route
+          path="*"
+          element={
+            <Navigate
+              to={isAuthenticated ? ROUTES.HOME : ROUTES.LOGIN}
+              replace
+            />
+          }
+        />
+      </Routes>
```

Comment:
```suggestion
         <Route path={ROUTES.LOGIN} element={<LoginPage />} />
        <Route element={<ProtectedRoutes userAuth={isAuthenticated} />}>
          <Route path={ROUTES.HOME} element={<ProjectsPage />} />
          <Route path="*" element={<Navigate to={ROUTES.HOME} replace />} />
        </Route>
        <Route path="*" element={<Navigate to={ROUTES.LOGIN} replace />} />
    
```

And then we can update it so